code/datastructure-set/src/test/java/ru/unn/agile/datastructure/set/model/MySetTest.java

```diff
@@ -0,0 +1,244 @@
+    public void canNotRemoveNullElement() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        mySet.remove(null);
+    }
+
+    @Test
+    public void canGiveIterator() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        Iterator<String> iterator = mySet.iterator();
+        Assert.assertNotNull(iterator);
+    }
+
+    @Test
+    public void canIterateByIterator() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        mySet.add("third row");
+        Iterator<String> iterator = mySet.iterator();
+        boolean foundValue = false;
+        while (iterator.hasNext()) {
+            if (iterator.next().equals("second row")) {
+                foundValue = true;
+            }
+        }
+
+        Assert.assertTrue(foundValue);
+    }
+
+    @Test
+    public void canRemoveByIterator() {
+        MySet<Integer> mySet = new MySet<>();
+        mySet.add(1);
+        mySet.add(2);
+        mySet.add(3);
+        Iterator<Integer> iterator = mySet.iterator();
+        while (iterator.hasNext()) {
+            if (iterator.next().equals(2)) {
+                iterator.remove();
+            }
+        }
+
+        Assert.assertEquals(mySet.size(), 2);
+    }
+
+    @Test
+    public void canCheckExistingCollection() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        mySet.add("third row");
+        Set<String> testSet = new HashSet<>();
+        testSet.add("first row");
+        testSet.add("second row");
+        Assert.assertTrue(mySet.containsAll(testSet));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void canNotCheckExistingCollectionWithNull() {
+        MySet<Integer> mySet = new MySet<>();
+        mySet.add(1);
+        mySet.add(2);
+        Set<Integer> testSet = new HashSet<>();
+        testSet.add(1);
+        testSet.add(null);
+        mySet.containsAll(testSet);
+    }
+
+    @Test
+    public void canRemoveCollectionsIntersection() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        Set<String> testSet = new HashSet<>();
+        testSet.add("first row");
+        Assert.assertTrue(mySet.removeAll(testSet));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void canNotRemoveNullCollection() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first row");
+        mySet.add("second row");
+        mySet.removeAll(null);
+    }
+
+    @Test
+    public void canUnionCollections() {
+        MySet<String> mySet = new MySet<>();
+        mySet.add("first");
+        Set<String> testSet = new HashSet<>();
+        testSet.add("second");
+        Assert.assertTrue(mySet.addAll(testSet));
```

Comment:
It seems to be a little bit strange to trust the returned boolean value. Please check here, are two collections equal

Reply:
equals will compare the whole collection, it can be a different test case. Test case is named canUnion. We validate the result in other test cases for add. Please see the code of add method and test for this. It can rise the duplications.